plugin/path_checkout_sets.go

```diff
@@ -21,6 +21,20 @@ type librarySet struct {
 	DisableCheckInEnforcement bool          `json:"disable_check_in_enforcement"`
 }
 
+// Validates ensures that a set meets our code assumptions that TTLs are set in
+// a way that makes sense, and that there's at least one service account.
+func (l *librarySet) Validate() error {
+	if len(l.ServiceAccountNames) < 1 {
+		return fmt.Errorf(`at least one service account must be configured`)
+	}
+	if l.MaxTTL > 0 {
+		if l.MaxTTL < l.TTL {
```

Comment:
Yeah! It would just make something essentially non-renewable.

Reply:
Is there a default ttl or default max ttl? And what exactly does "max ttl" mean _here_. I thought I've seen it other places to mean "the token/secret cannot live longer than this", but I may have also seen usage of "the maximum ttl cannot be greater than this", where the former is a finite limit to the life of the secret itself, and the later is a limit on the amount of time that can be used for a TTL. I hope that makes sense...

Reply:
I've also seen places where the default TTL is inherited from the max TTL, so that's why this stood out.

Reply:
I've only seen (or comprehended as) the former definition--how long can this secret/token/etc. be valid in total, including all renewal.

Reply:
Yeah, I was wondering too and I also was wondering if my code behaved the way I wanted, so I tested out how the leasing system behaves around this. Let me give you a couple of scenarios:
- The set's TTL is 0 and its maxTTL is 0: The resulting lease duration defaults to lasting 32 days because that's the default at the back-end level, which you have to tune to change.
- The set's TTL is 0 and its maxTTL is 10h: The resulting lease duration is 10h, and you won't be able to renew it.
- The set's TTL is 10h and its maxTTL is 20h: The resulting lease duration is 10h. If you renew it after 9h have passed, you'll get a secret with a lease duration of 10h again. If you renew _that_ after 9 hours have passed, you'll get a secret with a lease duration of 2h, because 9 + 9 + 2 = 20, the maxTTL.
- The set's TTL is 10h and its maxTTL is 10h: The resulting lease duration is 10h and you won't be able to renew it.

Reply:
Marking this as resolved because I think there's nothing else to do here.